BraintreeCore/src/main/java/com/braintreepayments/api/UserCanceledException.kt

```diff
@@ -0,0 +1,20 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+
+/**
+ * Error class thrown when a user cancels a payment flow
+ *
+ * @property isExplicitCancelation whether or not the user explicitly canceled the payment flow.
+ *
+ * This value will be true if the user manually confirms cancellation of the payment flow.
+ *
+ * This value will be false if the user returns to the app without completing the payment flow
+ * and the action performed to return to the app is unknown. For browser switching flows, this
+ * could mean the user returned to the app through multi-tasking without completing the flow,
+ * the user closed the browser tab, or the user pressed the back button.
+ */
+open class UserCanceledException @JvmOverloads @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP) constructor(
+    message: String?,
+    val isExplicitCancelation: Boolean = false
+) : BraintreeException(message)
```

Comment:
yeah this is what i meant with the above comment on `BraintreeException`

Reply:
Yeah that makes sense. The other ones were ones were good catches. We should leverage `@JvmOverload` when we can. At the root though we may want to consider an overhaul of error handling in `v5`.